eclair-core/src/main/scala/fr/acinq/eclair/db/sqlite/SqlitePaymentsDb.scala

```diff
@@ -33,14 +33,33 @@ class SqlitePaymentsDb(sqlite: Connection) extends PaymentsDb with Logging {
   import SqliteUtils.ExtendedResultSet._
 
   val DB_NAME = "payments"
-  val CURRENT_VERSION = 2
+  val CURRENT_VERSION = 3
 
-  using(sqlite.createStatement()) { statement =>
-    require(getVersion(statement, DB_NAME, CURRENT_VERSION) <= CURRENT_VERSION, s"incompatible version of $DB_NAME DB found") // version 2 is "backward compatible" in the sense that it uses separate tables from version 1. There is no migration though
+  def migration123(statement: Statement): Unit = {
+    // 1 -> 2 was backwards compatible so this method is used to migrate both 1 and 2 to 3
     statement.executeUpdate("CREATE TABLE IF NOT EXISTS received_payments (payment_hash BLOB NOT NULL PRIMARY KEY, preimage BLOB NOT NULL, payment_request TEXT NOT NULL, received_msat INTEGER, created_at INTEGER NOT NULL, expire_at INTEGER, received_at INTEGER)")
     statement.executeUpdate("CREATE TABLE IF NOT EXISTS sent_payments (id TEXT NOT NULL PRIMARY KEY, payment_hash BLOB NOT NULL, preimage BLOB, amount_msat INTEGER NOT NULL, created_at INTEGER NOT NULL, completed_at INTEGER, status VARCHAR NOT NULL)")
+    statement.executeUpdate("ALTER TABLE sent_payments ADD COLUMN fee_msat INTEGER DEFAULT 0 NOT NULL")
```

Comment:
Why not using a nullable field here?

Reply:
And use `feeMsat: Option[Long]` instead of `0` by default in `OutgoingPayment`?

Reply:
Right, having `feeMsat: Option[Long]` may be a stretch, the alternative would be to do the conversion in the db layer (keeping `feeMsat: Long` and nullable in the table) but that's even worse. I think we can keep it NOT NULL and use `0` as default value, after all if the payment wasn't successful we spent exactly `0` msat in fees.

Reply:
So no changes here then?

Reply:
Let's wait for more reviewers but for me is okay as it is, cc @sstone @pm47

Reply:
I don't have strong feelings about this but having it nullable forces the use of `Option[Long]` which is very correct but will lead to (IMHO) less elegant code where we need to check for the presence of the `feeMsat` on a finalized outgoing payment. On the other hand one could argue that using a default value of `0` is logically correct because a non finalized payment pays exactly `0` msat in fees.

Reply:
FWIW on a client side I'm working on using `0` is bit more comfortable as I can just `value + fee` instead of `value + feeOpt.getOrElse(0)`.